### PR TITLE
Michael/ir 4091 video

### PR DESCRIPTION
**Pull request by MbfloydIR**
## Summary
https://tsu.atlassian.net/browse/IR-4091
https://tsu.atlassian.net/browse/IR-4664
## References
closes #_insert number here_

### CODE REVIEW

File: packages/engine/src/scene/components/VideoComponent.tsx [hunk 1]
```diff
@@ -112,13 +112,36 @@ export const VideoComponent = defineComponent({
...
     fit: ContentFitTypeSchema('contain'),
+    useLetterBox: S.Bool(false),
     projection: ProjectionSchema,
     mediaUUID: S.EntityUUID(),
     // internal
     videoMeshEntity: S.Entity(),
     texture: S.Nullable(S.Type<VideoTexturePriorityQueue>())
   }),
 
+  onSet: (entity, component, json) => {
```
Comment (HexaField) [inline packages/engine/src/scene/components/VideoComponent.tsx]: i dont think we need the onset anymore

**Review by HexaField — CHANGES_REQUESTED** [review]

**Review by HexaField — DISMISSED** [review]
this is what the `fit` field should already accomplish. we should avoid making new fields and functions if `fit` is sufficient

File: packages/engine/src/scene/components/VideoComponent.tsx (hunk 1 above)
Comment (MbfloydIR) [reply packages/engine/src/scene/components/VideoComponent.tsx]: Talked with @MichaelEstes, the onSet is temporarily needed, other wise the video player doesn't work.

**Comment by MbfloydIR** [comment]
> this is what the `fit` field should already accomplish. we should avoid making new fields and functions if `fit` is sufficient

Removed the letterbox property and made the letterboxing effect always on

File: packages/engine/src/scene/components/VideoComponent.tsx (hunk 1 above)
Comment (MichaelEstes) [reply packages/engine/src/scene/components/VideoComponent.tsx]: Michael found an issue with component schemas for components that rely on three objects being cloned when set. The onSet can be removed when https://github.com/ir-engine/ir-engine/pull/285 is merged in

Comment (MbfloydIR) [reply packages/engine/src/scene/components/VideoComponent.tsx]: on set has now been removed

**Review by Rezmason — APPROVED** [review]
LGTM!

**Pull request merged**